src/map2world/map2world/map2world.py

```diff
@@ -3,132 +3,404 @@
         self.declare_parameter("box_height", 2.0)
-        
+        self.declare_parameter("package_name", "sim_models_ppmt")
+        self.declare_parameter("model_name", "new_model")
+        self.declare_parameter("img_path", "-1")
+        self.declare_parameter("map_mode", "clean")
+        self.declare_parameter("red", 255)
+        self.declare_parameter("green", 0)
+        self.declare_parameter("blue", 0)
+
         map_topic = self.get_parameter("map_topic").get_parameter_value().string_value
-        self.mesh_type = self.get_parameter("mesh_type").get_parameter_value().string_value
-        self.export_dir = self.get_parameter("export_dir").get_parameter_value().string_value
-        self.threshold = self.get_parameter("occupied_threshold").get_parameter_value().double_value
-        self.height = self.get_parameter("box_height").get_parameter_value().double_value
+        self.mesh_type = (
+            self.get_parameter("mesh_type").get_parameter_value().string_value
+        )
+        self.threshold = (
+            self.get_parameter("occupied_threshold").get_parameter_value().double_value
+        )
+        self.height = (
+            self.get_parameter("box_height").get_parameter_value().double_value
+        )
+        self.package_name = (
+            self.get_parameter("package_name").get_parameter_value().string_value
+        )
+        self.model_name = (
+            self.get_parameter("model_name").get_parameter_value().string_value
+        )
+        
+        self.img_path = (
+            self.get_parameter("img_path").get_parameter_value().string_value
+        )
+        
+        self.map_mode = (
+            self.get_parameter("map_mode").get_parameter_value().string_value
+        )
+        
+        self.red = self.get_parameter("red").get_parameter_value().integer_value
+        self.green = self.get_parameter("green").get_parameter_value().integer_value
+        self.blue = self.get_parameter("blue").get_parameter_value().integer_value
 
         map_sub_qos = QoSProfile(
             durability=QoSDurabilityPolicy.TRANSIENT_LOCAL,
             reliability=QoSReliabilityPolicy.RELIABLE,
             history=QoSHistoryPolicy.KEEP_LAST,
-            depth=1)
-
-        self.test_map_sub = self.create_subscription(OccupancyGrid, map_topic, self.map_callback, qos_profile=map_sub_qos)
+            depth=1,
+        )
 
+        self.test_map_sub = self.create_subscription(
+            OccupancyGrid, map_topic, self.map_callback, qos_profile=map_sub_qos
+        )
+        # self.export_dir = str(os.getcwd())
         # Probably there's some way to get trimesh logs to point to ROS
         # logs, but I don't know it.  Uncomment the below if something
         # goes wrong with trimesh to get the logs to print to stdout.
-        #trimesh.util.attach_to_log()
+        # trimesh.util.attach_to_log()
+        
+        self.template_path = "~/Documents/map2world/src/map2world/templates/"
```

Comment:
Why is this path hardcoded?

Reply:
Because relative path is not working. If generalised path is needed then i need to keep templates folder in share directory of map2world.